ui/raidboss/browser_tts_engine.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+class BrowserTTSEngine {
+  static play(text) {
+    try {
+      // check to see if we've already gotten this tts before
+      let iframe = window.document.querySelector('[data-tts="'+text+'"]');
+      if (iframe === null) {
+        iframe = document.createElement('iframe');
+        // remove sandbox so we can modify contents/call play on audio element later
+        iframe.removeAttribute('sandbox');
+        iframe.style.display = 'none';
+        document.body.appendChild(iframe);
+        let encText = encodeURIComponent(text);
+        iframe.contentDocument.body.innerHTML = '<audio src="https://translate.google.com/translate_tts?ie=UTF-8&client=tw-ob&tl=en&q=' + encText + '" id="TTS">';
```

Comment:
Support for window.speechSynthesis is spotty for basically anything other than Windows Chrome. I only tested on my own android device and an android SDK emulator, for instance, but neither had any voices to use so invoking the speak function didn't actually cause any noise. Since at least two members of my static would be relying on this working via mobile phone, I decided that the HTML5 audio option was the better option.

The weird way that the audio embed itself is written is to work around the Google service not sending CORS headers, and sending a 404 response if there's a Referer header on the request. I'm certainly open to other TTS options, I just went with the easiest/quickest to implement that wouldn't be making a new request every time the same TTS was used.

For Player HP, is that not available in the client via memory for the party? Or is the concern more that the HP decrease might be delayed due to network differences? Same for party player positions.

Reply:
https://codepen.io/matt-west/pen/wGzuJ worked fine on my Android device.  It looks like there's wide support for it on https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesis.

Re: HP.  It's not that it's not available, it's just that it's not sent for every player.

Reply:
Maybe some fallback logic for this? Check for window.speechSynthesis and valid voices and use that if available, otherwise fall back to this logic?

Reply:
To clarify my previous point about speechSynthesis, consider this output:
```
typeof window.speechSynthesis
"object"
window.speechSynthesis.getVoices().length
0
```

Reply:
Huh! What browser has that?

Reply:
Vivaldi (chromium) on Linux. Same thing with normal Chrome on my Linux distro, as well.